### PR TITLE
Logging Implementation

### PR DESCRIPTION
**Pull request by Root3287**
FIRST THINGS FIRST:
- INSTALLATION HAVE TO BE REVERTED
- Maybe remove the htmlpurifier document.

Logged most actions in the ACP, MCP, UCP, and forums. Still need to modify them to make sense to someone who is reading them... Especially on the forums. I will add the languages later. I'm heading off to bed :P. Can't code when tired.

The reason to have a full log is that the admin could track who is doing what. If a (possible)hacker takes over an account, the admins could possibly revert all the things that the expose account has done. Possibly. But I know this is not the development build just make sure the code is good before committing. Especially after a fresh install

 Changes to be committed:
	new file:   core/includes/htmlpurifier/standalone/HTMLPurifier/DefinitionCache/Serializer/HTML/4.7.0,241afe084734d0900affdf81b01044ae17792d23,1.ser
	modified:   core/installation/views/installer.view.php
	modified:   custom/languages/EnglishUK/admin.php
	new file:   custom/languages/EnglishUK/log.php
	modified:   custom/languages/EnglishUS/admin.php
	modified:   install.php
	modified:   modules/Core/init.php
	modified:   modules/Core/pages/admin/api.php
	modified:   modules/Core/pages/admin/auth.php
	modified:   modules/Core/pages/admin/groups.php
	modified:   modules/Core/pages/admin/images.php
	modified:   modules/Core/pages/admin/minecraft.php
	modified:   modules/Core/pages/admin/modules.php
	modified:   modules/Core/pages/admin/pages.php
	modified:   modules/Core/pages/admin/registration.php
	modified:   modules/Core/pages/admin/security.php
	modified:   modules/Core/pages/admin/styles.php
	modified:   modules/Core/pages/admin/users.php
	modified:   modules/Core/pages/admin/widgets.php
	modified:   modules/Core/pages/login.php
	modified:   modules/Core/pages/logout.php
	modified:   modules/Core/pages/mod/ip_lookup.php
	modified:   modules/Core/pages/mod/punishments.php
	modified:   modules/Core/pages/mod/reports.php
	modified:   modules/Core/pages/register.php
	modified:   modules/Core/pages/user/acknowledge.php
	modified:   modules/Core/pages/user/settings.php
	modified:   modules/Forum/pages/forum/delete.php
	modified:   modules/Forum/pages/forum/delete_post.php
	modified:   modules/Forum/pages/forum/edit.php
	modified:   modules/Forum/pages/forum/lock.php
	modified:   modules/Forum/pages/forum/merge.php
	modified:   modules/Forum/pages/forum/move.php
	modified:   modules/Forum/pages/forum/new_topic.php
	modified:   modules/Forum/pages/forum/reactions.php
	modified:   modules/Forum/pages/forum/report.php
	modified:   modules/Forum/pages/forum/search.php
	modified:   modules/Forum/pages/forum/stick.php
If there is any questions about this commit message me on my discord Root3287#2339 Thank you <3

### CODE REVIEW

File: modules/Core/pages/mod/punishments.php [hunk 1]
```diff
@@ -189,6 +192,25 @@
...
+                    $infractionTypeText = $language->get('mod', 'ban_ip');
+                }
+
+                $queries->create('logs', array(
+                        'time' => date('U'),
+                        'action' => $language->get('logs', 'log_puishment_revoke'),
+                        'user_id' => $user->data()->id,
+                        'ip' => $user->getIP(),
+                        'info' => $language->get('mod', 'punishment')."{$infractionTypeText}: ".$language->get('user', 'username').": ".Output::getClean(
+                            $queries->getWhere("users", ["id", "=", Output::getClean($_GET['user'])])[0]->username
```
Comment (samerton) [inline modules/Core/pages/mod/punishments.php]: Unfortunately,
```
Output::getClean($queries->getWhere("users", ["id", "=", Output::getClean($_GET['user'])])[0]->username)
```
is not compatible with PHP < 5.5 if I remember correctly, and Nameless is for 5.4+.

Can this be changed to:
```
$query_user = $queries->getWhere("users", ["id", "=", Output::getClean($_GET['user'])]);
$query_user = Output::getClean($query_user[0]->username);

$queries->create('logs', array(
  ......,
  'info' => $language->get('mod', 'punishment')."{$infractionTypeText}: ". $query_user,
```

**Comment by samerton** [comment]
Thanks for this! Would it be possible to check out my comment on the commit and make that (hopefully small) change?

**Comment by Root3287** [comment]
Okay, I was just skimming through the code and implemented that because I saw it here and there. Hopefully it’s a small change...

**Comment by Root3287** [comment]
I think it is committed into this pull request

**Comment by Root3287** [comment]
This is pretty much done. (That I believe) have a look and make sure everything is alright

**Pull request merged**